### DISCUSSION TITLE
realtime suscribe to specific "room"

### DISCUSSION THREAD

**Discussion by fernandoslim** [pocketbase/pocketbase#1716] [Q&A]
Hi, first I wanted to thank you for this tool, it's amazing!

I am currently writing a chat app, so I have 3 collections:

users, rooms, roomMessages

roomMessages got a relationship with users and rooms, so this is the structure: id, userId, roomId, message, ....

I've already everything working, but to suscribe I've done this: pb.collection("roomMessages").suscribe("*") as you can see I've suscrive to all collection and this is my question, How can suscribe to an spesific "room" (roomId) instead have to received all events and then filter it? 

Thank you in advance

**Comment by ganigeorgiev** [comment]
The realtime subscriptions doesn't support client-side defined filters at the moment and this will be eventually handled in #527 and #1657.

Since you are subscribing to the wildcard topic for the `roomMessages`, currently you can define only an admin-level filter using the "List API" rule of your collection (_aka. the client will receive create/update/delete change events only for the records they are allowed to list/search_) and then on the client-side you can filter manually the received event by inspecting the `e.record` props in the `subscribe` callback.
